File: bispo.py
import peca


class Bispo(peca.Peca):
    def __init__(self, posicao_origem_x, posicao_origem_y, cor):
        super().__init__(posicao_origem_x, posicao_origem_y, cor)

    def validar(self, posicao_x, posicao_y):
        if (posicao_x == self._posicao_atual_x) or (posicao_y == self._posicao_atual_y):
            return False
        if abs(posicao_y - self._posicao_atual_y) != abs(posicao_x - self._posicao_atual_x):
            return False
        return True

    def movimentar(self, destino_x, destino_y):
        if super().validar(destino_x, destino_y):
            if self.validar(destino_x, destino_y):
                print("Validar.... True")
                self._posicao_origem_x = self._posicao_atual_x
                self._posicao_origem_y = self._posicao_atual_y
                self._posicao_atual_x = destino_x
                self._posicao_atual_y = destino_y
                return True
            return False

    def desfazer_movimento(self):
        self._posicao_atual_x = self._posicao_origem_x
        self._posicao_atual_y = self._posicao_origem_y

File: src/xadrez_jv/Bispo.java
package xadrez_jv;

public class Bispo extends Peca{
	
	Bispo(int posicaoOrigemX, int posicaoOrigemY, int cor)
	{
		super(posicaoOrigemX, posicaoOrigemY, cor);
	}
	
	@Override
	public boolean validar(int posicaoX, int posicaoY)
	{
		if (posicaoX == this.posicaoAtualX || posicaoY == this.posicaoAtualY)
			return false;
		if (Math.abs(posicaoY - this.posicaoAtualY) != Math.abs(posicaoX - this.posicaoAtualX))
			return false;
		return true;
	}
	
	@Override 
	public boolean movimentar(int destinoX, int destinoY)
	{
		if (super.validar(destinoX, destinoY))
		{
			if (validar(destinoX, destinoY))
			{
				this.posicaoOrigemX = this.posicaoAtualX;
				this.posicaoOrigemY = this.posicaoAtualY;
				this.posicaoAtualX = destinoX;
				this.posicaoAtualY = destinoY;
				return true;
			}
		}
		return false;
	}
}
